### DISCUSSION TITLE
How can I customize hints marking while still using default matching?

### DISCUSSION THREAD

**Discussion by trygveaa** [kovidgoyal/kitty#6404] [Q&A]
I have a custom hints processor (https://github.com/trygveaa/kitty-hints-mouse-click) that has some custom code for marking, but also allows the default matching to be used. Previously (v0.27.1 and before, before the go rewrite) I imported `functions_for` and `regex_finditer` from `kittens.hints.main` to do this. Then I could use these two functions to generate the list of matches, and I could add my own matches in addition, and then I could yield `Mark` with a custom groupdict.

Now with v0.28.0 and later these functions are not available anymore since they're now in go. `regex_finditer` is pretty small and not specific to the different matching types, so that's easy enough to copy into the script. However, I rather not duplicate the `functions_for` function with all the logic for how to match against paths, words, urls etc. Is there any way I can get the list of matches from the script, so I can modify it and yield `Mark` myself?

**Comment by kovidgoyal** [comment]
No, I'm afraid that logic is not available anymore, you will need to
duplicate it in your script. You can just copy/paste it from kitty
source of 0.27.1
